Jenkinsfile

```diff
@@ -80,6 +81,33 @@ pipeline {
                                 '''
                             }
                         }
+                        stage("Static") {
+                            steps {
+                                sh '''
+                                docker run --rm -t --privileged \
+                                  -v "$WORKSPACE/bundles:/go/src/github.com/docker/docker/bundles" \
+                                  --name docker-pr$BUILD_NUMBER \
+                                  -e DOCKER_GITCOMMIT=${GIT_COMMIT} \
+                                  -e DOCKER_GRAPHDRIVER \
+                                  docker:${GIT_COMMIT} \
+                                  hack/make.sh binary-daemon
+                                '''
+                            }
+                        }
+                        stage("Cross") {
+                            steps {
+                                sh '''
+                                docker run --rm -t --privileged \
+                                  -v "$WORKSPACE/bundles:/go/src/github.com/docker/docker/bundles" \
+                                  --name docker-pr$BUILD_NUMBER \
+                                  -e DOCKER_GITCOMMIT=${GIT_COMMIT} \
+                                  -e DOCKER_GRAPHDRIVER \
+                                  docker:${GIT_COMMIT} \
+                                  hack/make.sh cross
+                                '''
+                            }
+                        }
+                        // needs to be last stage that calls make.sh for the junit report to work
```

Comment:
What was the problem if it wasn't the last stage? Does it help if you upload the junit.xml as part of the stage itself? (See https://github.com/moby/moby/pull/39719)

Reply:
The unit test stage needs to be last because the tests will be written to `bundles` directory and if the other stages run after the unit test stage then the directory would be wiped out.

Want to honor the return value of the unit tests, so if the tests has a failure then it will return non-zero back to the shell and jenkins will stop processing that step and mark that step as red. The `always` stage is an exception catch stage that will be run whether any of the steps fail, so that would be a suitable place to consume the junit.xml file.

Reply:
Does KEEPBUNDLE=1 work?

Reply:
Maybe. Can do in followup

Reply:
Each stage can upload its bundles/results before the next stage starts